aiida/schedulers/scheduler.py

```diff
@@ -283,20 +283,43 @@ def _get_detailed_jobinfo_command(self, jobid):
 
         :raises: :class:`aiida.common.exceptions.FeatureNotAvailable`
         """
-        # pylint: disable=no-self-use, not-callable, unused-argument
+        # pylint: disable=no-self-use,not-callable,unused-argument
         raise FeatureNotAvailable('Cannot get detailed job info')
 
+    def get_detailed_job_info(self, job_id):
+        """Return the detailed job info.
+
+        This will be a dictionary with the return value, stderr and stdout content returned by calling the command that
+        is returned by `_get_detailed_job_info_command`.
+
+        :param job_id: the job identifier
+        :return: dictionary with `retval`, `stdout` and `stderr`.
+        """
+        command = self._get_detailed_job_info_command(job_id)  # pylint: disable=assignment-from-no-return
+        with self.transport:
+            retval, stdout, stderr = self.transport.exec_command_wait(command)
+
+        detailed_job_info = {
+            'retval': retval,
+            'stdout': stdout,
+            'stderr': stderr,
+        }
+
+        return detailed_job_info
+
     def get_detailed_jobinfo(self, jobid):
         """
         Return a string with the output of the detailed_jobinfo command.
 
+        .. deprecated:: use the `get_detailed_job_info` instead
```

Comment:
Can we declare when we drop this (2.0) and issue a warning in the code?

Reply:
done